hack/source-manifests.sh

```diff
@@ -32,53 +32,66 @@ mkdir -p $OUTDIR_CRDS
-	($IMAGE_RUN_CMD --entrypoint=cat "$ROOK_IMAGE" "$crd_file") > $OUTDIR_CRDS/"$(basename "$crd_file")"
-done;
-rm -f "$crd_list"
+function dump_rook_csv() {
+	rook_template_dir="/etc/ceph-csv-templates"
+	rook_csv_template="rook-ceph-ocp.vVERSION.clusterserviceversion.yaml.in"
+	rook_crds_dir=$rook_template_dir/crds
+	crd_list=$(mktemp)
+	echo "Dumping rook csv using command: $IMAGE_RUN_CMD --entrypoint=cat $ROOK_IMAGE $rook_template_dir/$rook_csv_template"
+	$IMAGE_RUN_CMD --entrypoint=cat "$ROOK_IMAGE" $rook_template_dir/$rook_csv_template > $ROOK_CSV
+	echo "Listing rook crds using command: $IMAGE_RUN_CMD --entrypoint=ls $ROOK_IMAGE -1 $rook_crds_dir/"
+	$IMAGE_RUN_CMD --entrypoint=ls "$ROOK_IMAGE" -1 $rook_crds_dir/ > "$crd_list"
+	# shellcheck disable=SC2013
+	for i in $(cat "$crd_list"); do
+	        # shellcheck disable=SC2059
+		crd_file=$(printf ${rook_crds_dir}/"$i" | tr -d '[:space:]')
+		echo "Dumping rook crd $crd_file using command: $IMAGE_RUN_CMD --entrypoint=cat $ROOK_IMAGE $crd_file"
+		($IMAGE_RUN_CMD --entrypoint=cat "$ROOK_IMAGE" "$crd_file") > $OUTDIR_CRDS/"$(basename "$crd_file")"
+	done;
+	rm -f "$crd_list"
+}
 
 # ==== DUMP OCS YAMLS ====
 # Generate an OCS CSV using the operator-sdk.
 # This is the base CSV everything else gets merged into later on.
-TMP_CSV_VERSION=9999.9999.9999
-gen_args="generate csv --csv-version=$TMP_CSV_VERSION --output-dir=$OUTDIR_TEMPLATES"
-if [ -n "$CSV_REPLACES_VERSION" ]; then
-	gen_args="$gen_args --from-version=$CSV_REPLACES_VERSION"
-fi
-# shellcheck disable=SC2086
-$OPERATOR_SDK $gen_args
-mv $OUTDIR_TEMPLATES/manifests/ocs-operator.clusterserviceversion.yaml $OCS_CSV
-# Make variables templated for csv-merger tool
-if [ "$OS_TYPE" == "Darwin" ]; then
-	sed -i '' "s/$TMP_CSV_VERSION/{{.OcsOperatorCsvVersion}}/g" $OCS_CSV
-	sed -i '' "s/REPLACE_IMAGE/{{.OcsOperatorImage}}/g" $OCS_CSV
-	sed -i '' "/replaces:/d" $OCS_CSV
-else
-	sed -i "s/$TMP_CSV_VERSION/{{.OcsOperatorCsvVersion}}/g" $OCS_CSV
-	sed -i "s/REPLACE_IMAGE/{{.OcsOperatorImage}}/g" $OCS_CSV
-	sed -i "/replaces:/d" $OCS_CSV
+function gen_ocs_csv() {
+	TMP_CSV_VERSION=9999.9999.9999
+	gen_args="generate csv --csv-version=$TMP_CSV_VERSION --output-dir=$OUTDIR_TEMPLATES"
+	if [ -n "$CSV_REPLACES_VERSION" ]; then
+		gen_args="$gen_args --from-version=$CSV_REPLACES_VERSION"
+	fi
+	# shellcheck disable=SC2086
+	$OPERATOR_SDK $gen_args
+	mv $OUTDIR_TEMPLATES/manifests/ocs-operator.clusterserviceversion.yaml $OCS_CSV
+	# Make variables templated for csv-merger tool
+	if [ "$OS_TYPE" == "Darwin" ]; then
+		sed -i '' "s/$TMP_CSV_VERSION/{{.OcsOperatorCsvVersion}}/g" $OCS_CSV
+		sed -i '' "s/REPLACE_IMAGE/{{.OcsOperatorImage}}/g" $OCS_CSV
+		sed -i '' "/replaces:/d" $OCS_CSV
+	else
+		sed -i "s/$TMP_CSV_VERSION/{{.OcsOperatorCsvVersion}}/g" $OCS_CSV
+		sed -i "s/REPLACE_IMAGE/{{.OcsOperatorImage}}/g" $OCS_CSV
+		sed -i "/replaces:/d" $OCS_CSV
+	fi
+	cp deploy/crds/* $OUTDIR_CRDS/
+}
+
+if [ ! -n "$OPENSHIFT_BUILD_NAMESPACE" ]; then
```

Comment:
why `if [ ! -n "..." ]` instead of `if [ -z "..." ]` ?

Reply:
heh, shellcheck confirms this: `^-- SC2236: Use -z instead of ! -n.`

Reply:
D'oh! Fair.